### PR TITLE
Organized the frontend a bit more... 🤔

### PR DESCRIPTION
**Pull request by indefinities**
- added [React Material UI](https://mui.com/)
- removed all `creat-react-app` files from the `pokedex` folder
- added `TrainersList` into `App.js`
- created `trainer-service` and `trainer-list`
- ... so I don't know what to write to obtain the data from the backend lol help 😭

### CODE REVIEW

File: src/trainers/trainer-service.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+// declare URL where server listens for HTTP requests
+const TRAINERS_URL = "http://localhost:8080/api/trainers"
```
Comment (CookieComputing) [inline src/trainers/trainer-service.js:2]: is your port on 8080? The backend should be configured to 8000

Comment (CookieComputing) [reply src/trainers/trainer-service.js:2]: follow the api endpoints to a t: `"http://localhost:8080/trainers"

Comment (indefinities) [reply src/trainers/trainer-service.js:2]: nope this is from the homework, thanks for the clarification

File: src/trainers/trainer-service.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+// retrieve all users from the server
+export const findAllTrainers = () =>
+    fetch(TRAINERS_URL)
+    .then(response => response.json())
+
+
+// retrieve a single Trainerby their ID
+export const findTrainerById = (id) =>
+    fetch(`${TRAINERS_URL}/${id}`)
```
Comment (CookieComputing) [inline src/trainers/trainer-service.js:12]: Forgot to mention this in the api docs, but make sure everything ends in a `/` (trailing slash) e.g.  ``${TRAINERS_URL}/${id}/`. Django doesn't like the non-trailing slash unless I specifically add a setting to do so, but that can come in a future PR

File: src/trainers/trainer-service.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    .then(response => response.json())
+
+
+// delete a Trainerby their ID
+export const deleteTrainer = (id) =>
+    fetch(`${TRAINERS_URL}/${id}`, {
```
Comment (CookieComputing) [inline src/trainers/trainer-service.js:18]: same as above

File: src/trainers/trainer-service.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+export const createTrainer= (user) =>
+    fetch(TRAINERS_URL, {
+      method: 'POST',
+      body: JSON.stringify(user),
+      headers: {'content-type': 'application/json'}
+    })
+    .then(response => response.json())
+
+
+// update a Trainerby their ID
```
Comment (CookieComputing) [inline src/trainers/trainer-service.js:33]: nit: "Trainer by"

File: src/trainers/trainer-list.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+const TrainersList = () => {
+    // const history = useHistory();
+    // const [trainers, setTrainers] = useState([]);
+    // useEffect(() => {
+    //   findAllUsers()
+    // }, [])
+    // const findAllUsers = () =>
+    //     trainerService.findAllUsers()
+    //     .then(trainers => setTrainers(trainers))
```
Comment (amywhying) [inline src/trainers/trainer-list.js:16]: AHHH THE COMMENTS
also everything looks good but I'd change the file name to be the same as the function name - i.e. `trainers-list.js` or `trainersList.js` whichever you prefer

**Review by amywhying — APPROVED** [review]

**Review by CookieComputing — CHANGES_REQUESTED** [review]
couple of comments on the api fetching process

File: src/trainers/trainer-service.js (hunk 1 above)
Comment (CookieComputing) [reply src/trainers/trainer-service.js:2]: ye, you can use the http endpoints provided in the [Trainer section](https://github.com/CookieComputing/db-pokedex-backend#trainer-api) as well as the move and pokemonInfo sections

File: src/trainers/trainer-list.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+function TrainersList() {
+
+  const [trainers, setTrainers] = useState([])
+  useEffect(() => {
+    findAllTrainers()
+  }, [])
+
+  const findAllTrainers = () =>
+    trainerService.findAllTrainers()
+    .then(trainers => setTrainers(trainers))
```
Comment (indefinities) [inline src/trainers/trainer-list.js:16]: @CookieComputing pls send help

Comment (CookieComputing) [reply src/trainers/trainer-list.js:16]: look at my comment below for your return value

File: src/trainers/trainer-service.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+export const findTrainerById = (id) =>
+    fetch(`${TRAINERS_URL}/${id}/`)
+    .then(response => response.json())
+
+
+// delete a Trainer by their ID
+export const deleteTrainer = (id) =>
+    fetch(`${TRAINERS_URL}/delete/${id}/`, {
+        method: 'POST',
+        body: JSON.stringify(id),
```
Comment (CookieComputing) [inline src/trainers/trainer-service.js:20]: we don't expect a body in delete on the backend, you can remove this line

**Review by CookieComputing — CHANGES_REQUESTED** [review]
see feedback

File: src/trainers/trainer-service.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
+// declare URL where server listens for HTTP requests
+const TRAINERS_URL = "http://localhost:8000/trainers"
+
+// retrieve all users from the server
+export const findAllTrainers = () =>
+    fetch(TRAINERS_URL)
+    .then(response => console.log(response.json()))
```
[the remark this answers is not in the document]
Comment (CookieComputing) [reply src/trainers/trainer-service.js:7]: also pls remove `console.log()` before PR merge

**Pull request closed without merging**